src/BODSDataExtractor/extractor.py

```diff
@@ -78,23 +78,24 @@ def check_api_response(self, response,j1):
         #if response is less than 4, no data has been pulled
         
         if len(j1)<4:
-            print("_____________________________________")
-            print("Error")
+            message=""
             #we are extracting the status code (key) and the reason (value) 
             
-            
+            #checking through items in json api response file dictionary
             for key,value in j1.items():
-                print(key, ':', value)
-            print("_____________________________________")
-            sys.exit()
+                content=str(key) +" : "+ str(value)
+                
+
+                message="\n"+message+str(content)+"\n"
+                
+                
+            raise ValueError(message)
+
             
         #if the api key is valid but the dataset is empty
         elif j1.get("results")==[]:
-            print("_____________________________________")
-            print("Error")
-            print("status_code : 200 OK")
-            print('reason : {"detail":"Empty Dataset"}')
-            sys.exit()
+            
+            raise ValueError('\n status_code : 200 OK \n reason : {"Empty Dataset"}')

```

Comment:
Better to return the actual status code from the response as a variable, rather than code it into the message here. Just in case something weird happens

Reply:
same with line 103 below, better to go and get the response code from the api response

Reply:
![image](https://user-images.githubusercontent.com/114913914/211346063-e86cf9c8-8a36-42e3-951a-d3c83de7e82b.png)
When an empty dataset is entered, the response doesn't actually contain a status code or reason as shown above.